### DISCUSSION TITLE
Can anyone confirm about ColorScheme in Storybook (v8) without storybook-dark-mode?

### DISCUSSION THREAD

**Discussion by shin-sforzando** [mantinedev/mantine#5906] [Q&A] [answered]
While reading https://mantine.dev/guides/storybook/ and upgrading to Storybook v8, I wondered if we really need to add [storybook-dark-mode](https://storybook.js.org/addons/storybook-dark-mode).
[@storybook/addon-backgrounds](https://storybook.js.org/addons/@storybook/addon-backgrounds) is part of [essentials](https://storybook.js.org/docs/react/essentials) and so is installed in all new Storybooks by default.

So I tried writing `.storybook/preview.tsx` as follows and it seems to work well.

```.storybook/preview.tsx
import "@mantine/core/styles.css"

import { MantineProvider, useMantineColorScheme } from "@mantine/core"

import type { Preview } from "@storybook/react"
import React, { type ReactNode, useEffect } from "react"

function ColorSchemeWrapper({
  children,
  backgrounds,
}: Readonly<{ children: ReactNode; backgrounds?: { value: string } }>) {
  const { setColorScheme } = useMantineColorScheme()

  useEffect(() => {
    if (backgrounds) {
      const red = backgrounds.value.substring(1, 3)
      const green = backgrounds.value.substring(3, 5)
      const blue = backgrounds.value.substring(5, 7)
      const brightness =
        (Number.parseInt(red, 16) * 299 +
         Number.parseInt(green, 16) * 587 +
         Number.parseInt(blue, 16) * 114) / 1000
      setColorScheme(128 <= brightness ? "light" : "dark")
    }
  }, [backgrounds, setColorScheme])

  return <>{children}</>
}

const preview: Preview = {
  decorators: [
    (Story, context) => {
      return (
        <MantineProvider>
          <ColorSchemeWrapper backgrounds={context.globals.backgrounds}>
            <Story />
          </ColorSchemeWrapper>
        </MantineProvider>
      )
    },
  ],
}

export default preview
```

I'm a React newbie so I'm sure there are better ways to write it.
With that said, can anyone point out if there are any problems with the above approach?

![Screenshot 2024-03-12 at 19 02 44](https://github.com/mantinedev/mantine/assets/32637762/f620833c-58ea-45b6-a776-7c2fae6d94d1)

My Environment:

```package.json
  "volta": {
    "node": "21.7.1",
    "npm": "10.2.4"
  },
  "dependencies": {
    "@mantine/core": "^7.6.1",
    "@mantine/hooks": "^7.6.1",
    "next": "14.1.3",
    "react": "^18",
    "react-dom": "^18"
  },
  "devDependencies": {
    "@storybook/addon-a11y": "^8.0.0",
    "@storybook/addon-coverage": "^1.0.1",
    "@storybook/addon-essentials": "^8.0.0",
    "@storybook/addon-interactions": "^8.0.0",
    "@storybook/addon-links": "^8.0.0",
    "@storybook/blocks": "^8.0.0",
    "@storybook/nextjs": "^8.0.0",
    "@storybook/react": "^8.0.0",
    "@storybook/test": "^8.0.0",
    "@storybook/test-runner": "^0.17.0",
    "@testing-library/react": "^14.2.1",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "chromatic": "^11.0.8",
    "jsdom": "^24.0.0",
    "postcss": "^8.4.35",
    "postcss-preset-mantine": "^1.13.0",
    "postcss-simple-vars": "^7.0.1",
    "storybook": "^8.0.0",
    "typescript": "^5",
  }
```

----

P.S.
Yes, I actually didn't add [@storybook/addon-styling-webpack](https://storybook.js.org/addons/@storybook/addon-styling-webpack), even though I don't use Vite.
I'm not sure what addon-styling-webpack is, but it's working fine.
It probably has nothing to do with [storybook-dark-mode](https://storybook.js.org/addons/storybook-dark-mode), but if anyone knows why it works without @storybook/addon-styling-webpack, please let me know.

**Comment by rtivital** [comment, marked as answer]
- backgrounds addon does not change color scheme property and adds background color that is not defined by Mantine or your application. In most cases, this behavior is unwanted.
- addon-styling-webpack is required for your `*.module.css` files to work as described in documentation. If you are not using PostCSS with mantine preset, then it is not required.

**Reply by shin-sforzando** [reply]
> addon-styling-webpack is required for your *.module.css files to work as described in documentation. If you are not using PostCSS with mantine preset, then it is not required.

Thanks for your quick & nice answer! I was convinced.

However, this is not yet well understood.

> backgrounds addon does not change color scheme property and adds background color that is not defined by Mantine or your application.

Is storybook-dark-mode setting the properties correctly for Mantine?

**Reply by rtivital** [reply]
Yes, as mentioned in the guide, it hooks up to the MantineProvider and changes Mantine color scheme.

**Reply by shin-sforzando** [reply]
I really appreciate the many times you have answered my questions.
And I apologize for my poor explanation.

My approach is also using `setColorScheme()`, so I can say I'm hooking into MantineProvider.
The reason why I tried this approach is that Storybook has evolved to v8, while most addons have not been able to keep up.
The same is expected to happen for future updates.
cf. https://github.com/storybookjs/storybook/issues/26031

So my motivation was to reduce external dependencies as much as possible if I could do the same thing, and even if I had no choice but to depend on it, I wanted to depend on something as close to official as possible.

It's easier to use `story-dark-mode`, that's for sure. Therefore I'm not considering making my approach a Pull Request or anything like that.
To rephrase my question, "Is this a good enough way to update Mantine's ColorScheme, and is there anything else I should be aware of?" That was the intent.

**Reply by rtivital** [reply]
story-dark-mode is the best solution I know about. Of course, in your application you can use whatever logic you prefer if it works for you

**Reply by shin-sforzando** [reply]
Thank you. Trying it out and understanding Mantine a little better is good enough for me.
